apps/mux/src/config.tsx

```diff
@@ -0,0 +1,295 @@
+/* eslint-disable jsx-a11y/media-has-caption */
+
+import React from 'react';
+
+import '@contentful/forma-36-react-components/dist/styles.css';
+import '@contentful/forma-36-fcss/dist/styles.css';
+import {
+  Heading,
+  Subheading,
+  Note,
+  Form,
+  TextField,
+  CheckboxField,
+  Paragraph,
+  Typography,
+  FieldGroup,
+} from '@contentful/forma-36-react-components';
+import {
+  AppExtensionSDK,
+  AppConfigAPI,
+  SpaceAPI,
+  BaseExtensionSDK,
+} from 'contentful-ui-extensions-sdk';
+import {
+  editorInterfacesToSelectedFields,
+  getCompatibleFields,
+  selectedFieldsToTargetState,
+  EditorInterface,
+  ContentType,
+  CompatibleFields,
+  SelectedFields,
+} from 'shared-dam-app';
+import MuxLogoSvg from './mux-logo.svg';
+import './config.css';
+
+interface ConfigProps {
+  sdk: AppExtensionSDK;
+}
+
+interface IParameters {
+  muxAccessTokenId?: string;
+  muxAccessTokenSecret?: string;
+}
+
+interface IState {
+  parameters: IParameters;
+  contentTypes: ContentType[];
+  compatibleFields: CompatibleFields;
+  selectedFields: SelectedFields;
+}
+
+class Config extends React.Component<ConfigProps, IState> {
+  app: AppConfigAPI;
+  space: SpaceAPI;
+
+  constructor(props: ConfigProps) {
+    super(props);
+    this.state = {
+      parameters: {},
+      contentTypes: [],
+      compatibleFields: {},
+      selectedFields: {},
+    };
+
+    // `sdk.app` exposes all app-related methods.
+    this.app = this.props.sdk.app;
+    this.space = this.props.sdk.space;
+
+    // `onConfigure` allows to configure a callback to be
+    // invoked when a user attempts to install the app or update
+    // its configuration.
+    this.app.onConfigure(() => this.onConfigure());
```

Comment:
Could you move this to `componentDidMount`? While testing, I saw some flakiness with this call and I think it may have to do with it being here in the constructor.